### PR TITLE
[Fix](multi catalog)Fix partition case bug.

### PR DESCRIPTION
**Pull request by Jibing-Li**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Set column names from path to lower case in case-insensitive case.
This is for Iceberg columns from path. Iceberg columns are case sensitive, which may cause error for table with partitions.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.88 seconds
 stream load tsv:          468 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         35 seconds loaded 2358488459 Bytes, about 64 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230214154351_clickbench_pr_96442.html

File: fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java [hunk 1]
```diff
@@ -119,6 +119,12 @@ public static List<String> parseColumnsFromPath(
         if (columnsFromPath == null || columnsFromPath.isEmpty()) {
             return Collections.emptyList();
         }
+        if (!caseSensitive) {
```
Comment (carlvinhust2012) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java:122]: why not add some case in UT or regression test?

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/common/util/BrokerUtil.java:122]: Will add some test cases.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**